corporate/migrations/0043_remove_customer_default_discount_and_more.py

```diff
@@ -0,0 +1,76 @@
+# Generated by Django 5.0.5 on 2024-05-03 06:50
+
+from decimal import Decimal
+
+from django.db import migrations, models
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+from corporate.lib.stripe import get_price_per_license
+
+
+def calculate_discounted_price_per_license(
+    original_price_per_license: int, discount: Decimal
+) -> int:
+    # There are no fractional cents in Stripe, so round down to nearest integer.
+    return int(float(original_price_per_license * (1 - discount / 100)) + 0.00001)
+
+
+def calculate_discounted_price(apps: StateApps, schema_editor: BaseDatabaseSchemaEditor) -> None:
+    BILLING_SCHEDULE_ANNUAL = 1
+    BILLING_SCHEDULE_MONTHLY = 2
+
+    Customer = apps.get_model("corporate", "Customer")
+    customers_to_update = []
+    for customer in Customer.objects.all():
+        if not customer.required_plan_tier or not customer.default_discount:
+            continue
+
+        annual_price_per_license = get_price_per_license(
+            customer.required_plan_tier, BILLING_SCHEDULE_ANNUAL
+        )
+        customer.annual_discounted_price = calculate_discounted_price_per_license(
+            annual_price_per_license, customer.default_discount
+        )
+        monthly_price_per_license = get_price_per_license(
+            customer.required_plan_tier, BILLING_SCHEDULE_MONTHLY
+        )
+        customer.monthly_discounted_price = calculate_discounted_price_per_license(
+            monthly_price_per_license, customer.default_discount
+        )
+        customers_to_update.append(customer)
+
+    Customer.objects.bulk_update(
+        customers_to_update, ["annual_discounted_price", "monthly_discounted_price"]
+    )
```

Comment:
So this seems to only migrate values for `Customer` objects, not for plans; is the reason that plans already have resolved `price_per_license` values and billing schedules, and thus are not directly impacted by this migration?

Would be good to document that in a docstring on this function.

Reply:
Added such a docstring.